### ISSUE TITLE
genbil adding attributes "column-index" and "type" to <expression> tag

### ISSUE THREAD

**Issue by allihandra** [Seddryck/NBi#401]
I have a template that has an expression tag, when the template is populated it now includes column-index and type attributes which are not part of the .xsd and cause tests suite to NOT open. using 1.18
I was using your sample code, can you see any issue with how I did this?
template:
```
<test name="$trunc(alias); separator=" and "$ to calculate $last(alias)$">
    <system-under-test>
      <resultSet>
        <query>
          <![CDATA[
          select
          {
         $measure; separator=", 
"$
          } on 0,
          {
          [Dim Client].[ClientState].members
          } on 1
          from
          [$perspective$]
          ]]>
        </query>
      </resultSet>
    </system-under-test>
    <assert>

 <all-rows>
        $alias:{ x | <alias column-index="$i$">$x$</alias>}$
        <expression name="calculate">
           Round($expression$, $round$) = Round($last(alias)$, $round$)
        </expression>
        <predicate operand="calculate" type="boolean">
          <true />
        </predicate>
      </all-rows>
    </assert>
  </test>
``` 

Output with settings removed:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testSuite xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://NBi/TestSuite">
   <test name="charge and budget to calculate variance">
    <system-under-test>
      <resultSet>
        <query><![CDATA[
          select
          {
         [Measures].[Charge Hours], 
         [Measures].[BudgetHours], 
         [Measures].[Budget Variance]
          } on 0,
          {
          [Dim Client].[ClientState].members
          } on 1
          from
          [CurrentSummary]
          ]]></query>
      </resultSet>
    </system-under-test>
    <assert>
      <all-rows>
        <expression name="calculate" column-index="0" type="text">
           Round(charge-budget, 2) = Round(variance, 2)
        </expression>
        <alias column-index="1">charge</alias>
        <alias column-index="2">budget</alias>
        <alias column-index="3">variance</alias>
        <predicate operand="calculate" type="boolean">
          <true />
        </predicate>
      </all-rows>
    </assert>
  </test>
</testSuite>
```

In case the genbiL steps have caused the issue I am including for reference:

```
case scope 'calcs';
case load file '..\Parameters\calculations.csv';
case scope 'env';
case load file '..\Configurations\CUR\schema.csv';
case scope 'cur';
case cross 'calcs' with 'env' on 'schema';

case split column 'alias' with value ',';
case split column 'measure' with value ',';
template load file '..\TestProject\DATA\MEASURES\METADATA\_templates\calcmember-validatexpression.nbitt';
suite generate;
setting include file '..\Configurations\CUR\settings.nbisettings';
suite save '..\TestProject\DATA\MEASURES\METADATA\CUR\calcmember-validatexpression.nbits';
```

**Comment by Seddryck** [comment]
This is a bug. I fixed something relatively close in #392. You should try the same beta because it could potentially fix your issue (but I doubt). 

I will take a look to this issue when I’ll be back from holidays next week. (And also ensure that *alias* children are before the *expression* children).

**Comment by Seddryck** [comment]
I saw that you forked the project. If you want to fix this by yourself the issue is in the serialisation (project NBi.Xml folder *items* then *calculations*). There you have a file *ExpressionXml* with an attribute *column-index*. The problem is probably that at some places an expression requires a *column-index* but not where you are using it. You should duplicate the class with inheritance and use the *light* version for your case.

**Comment by allihandra** [comment]
No I am really new at doing anything besides reading github, I didnt mean to create a fork, I will look at how to un-do it, sorry about that :)

**Comment by Seddryck** [comment]
The fact that you've an unexpected ```column-index``` is a duplicate of #372 and it's already fixed and available in the last beta (see #372 to get it).

The fact that you've an unexpected ```type``` is not a bug. ```expression``` is requiring a *type* and the default value (not serialized) is *text*.

The fact that ```expression``` was serialized before ```alias``` was an error and is now fixed in https://ci.appveyor.com/project/Seddryck/nbi/build/1.19.0-beta.47.build.441
